server.py

```diff
@@ -50,6 +51,67 @@ def get_shoes(self):
         conn.close()
         return jsonify(data)
     
+    def post_shoes(self):
+        conn = self.__get_db_connection()
+        cursor = conn.cursor()
+
+        #     cursor.execute('''
+        #     CREATE TABLE IF NOT EXISTS shoes (
+        #         shoes_id INTEGER PRIMARY KEY,
+        #         brand TEXT NOT NULL,
+        #         shoe_name TEXT NOT NULL,
+        #         shape TEXT NOT NULL,
+        #         size INT NOT NULL,
+        #         menwear TEXT NOT NULL,
+        #         womanwear TEX NOT NULL,
+        #         price INT NOT NULL,
+        #         manufacturer TEXT NOT NULL      
+        #     )
+        # ''')
+        data = request.get_json()
+        #print(data)
+        brand = data['brand']
+        #print(brand)
+        shoe_name = data['shoe_name']
+        #print(shoe_name)
+        shape = data['shape']
+        #print(shape)
+        size = data['size']
+        #print(size)
+        menwear = data['menwear']
+        #print(menwear)
+        womanwear = data['womanwear']
+        #print(womanwear)
+        price = data['price']
+        #print(price)
+        manufacturer = data['manufacturer']
+        #print(manufacturer)
+
+        checklist = {
+            "brand" : brand,
+            "shoe_name" : shoe_name,
+            "shape" : shape,
+            "size" : size,
+            "menwear" : menwear,
+            "womanwear" : womanwear,
+            "price" : price,
+            "manufacturer" : manufacturer
+        }
+
+        for shoeinfo, shoevalue in checklist.items():
```

Comment:
This validation should happen before creating checklist dict. Validations should always be the first step. It is because consider this scenario where _brand_ is missing from the request body, in that case data["brand"] will create panic and return KeyError. So, before accessing this data, validation logic should take care of whether it is present or not.